include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -853,73 +846,42 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::reinit(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const ArrayView<const Point<dim>> &                         unit_points)
 {
-  // If using the fast path, we need to precompute the mapping data.
-  if (!poly.empty())
-    internal::FEPointEvaluation::compute_mapping_data_for_generic_points(
-      *mapping, cell, unit_points, update_flags_mapping, mapping_data);
-
-  // then call the other version of this function with the precomputed data
-  reinit(cell, unit_points, mapping_data);
-}
-
+  // reinit is only allowed for mapping computation on the fly
+  AssertThrow(precomputed_mapping == false, ExcNotImplemented());
 
+  if (fast_path)
+    {
+      mapping_info_on_the_fly =
+        std::make_unique<NonMatching::MappingInfo<dim, spacedim>>(*mapping,
+                                                                  update_flags);
 
-template <int n_components, int dim, int spacedim, typename Number>
-void
-FEPointEvaluation<n_components, dim, spacedim, Number>::reinit(
-  const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-  const ArrayView<const Point<dim>> &                         unit_points,
-  const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
-                                                                     spacedim>
-    &precomputed_mapping_data)
-{
-  this->unit_points.resize(unit_points.size());
-  std::copy(unit_points.begin(), unit_points.end(), this->unit_points.begin());
+      mapping_info = mapping_info_on_the_fly.get();
```

Comment:
Then you could call this in the constructor and remove the `precomputed_mapping` variable.